File: Binary_Heap/max_binary_heap_insert.c
/* File: max_binary_heap_insert.c */

#include <stdio.h>

#define MAX 100

int heap[MAX];
int size = 0;

/* Function to swap two numbers */
void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

/* Insert element into Max Heap */
void insert(int value)
{
    size++;
    int i = size;

    heap[i] = value;

    /* Heapify up */
    while (i > 1 && heap[i] > heap[i / 2])
    {
        swap(&heap[i], &heap[i / 2]);
        i = i / 2;
    }
}

/* Display Heap */
void printHeap()
{
    for (int i = 1; i <= size; i++)
        printf("%d ", heap[i]);
    printf("\n");
}

int main()
{
    insert(20);
    insert(15);
    insert(30);
    insert(40);
    insert(10);

    printf("Max Heap Elements:\n");
    printHeap();

    return 0;
}

File: Binary_Heap/MaxBinaryHeapInsert.java
// File: MaxBinaryHeapInsert.java

class MaxBinaryHeapInsert {

    int[] heap = new int[100];
    int size = 0;

    void insert(int value) {
        size++;
        int i = size;

        heap[i] = value;

        while (i > 1 && heap[i] > heap[i / 2]) {
            int temp = heap[i];
            heap[i] = heap[i / 2];
            heap[i / 2] = temp;

            i = i / 2;
        }
    }

    void printHeap() {
        for (int i = 1; i <= size; i++)
            System.out.print(heap[i] + " ");
        System.out.println();
    }

    public static void main(String[] args) {

        MaxBinaryHeapInsert h = new MaxBinaryHeapInsert();

        h.insert(20);
        h.insert(15);
        h.insert(30);
        h.insert(40);
        h.insert(10);

        System.out.println("Max Heap Elements:");
        h.printHeap();
    }
}
